lib/fileutils.rb

```diff
@@ -501,25 +501,81 @@ def cp_lr(src, dest, noop: nil, verbose: nil,
   end
   module_function :cp_lr
 
+  # Creates {symbolic links}[https://en.wikipedia.org/wiki/Symbolic_link].
   #
-  # :call-seq:
-  #   FileUtils.ln_s(target, link, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln_s(target,  dir, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln_s(targets, dir, force: nil, noop: nil, verbose: nil)
+  # When +src+ is the path to an existing file:
+  #
+  # - When +dest+ is the path to a non-existent file,
+  #   creates a symbolic link at +dest+ pointing to +src+:
+  #
+  #     FileUtils.touch('src0.txt')
+  #     File.exist?('dest0.txt')   # => false
+  #     FileUtils.ln_s('src0.txt', 'dest0.txt')
+  #     File.symlink?('dest0.txt') # => true
+  #
+  # - When +dest+ is the path to a directory,
+  #   creates a symbolic link at <tt>dest/src</tt> pointing to +src+:
+  #
+  #     FileUtils.touch('src1.txt')
+  #     FileUtils.mkdir('destdir1')
+  #     FileUtils.ln_s('src1.txt', 'destdir1')
+  #     File.symlink?('destdir1/src1.txt') # => true
+  #
+  # - When +dest+ is the path to an existing file,
+  #   creates a symbolic link at +dest+ pointing to +src+
+  #   if and only if keyword argument <tt>force: true</tt> is given
+  #   (raises an exception otherwise):
+  #
+  #     FileUtils.touch('src2.txt')
+  #     FileUtils.touch('dest2.txt')
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt', force: true)
+  #     FileTest.symlink?('dest2.txt') # => true
+  #
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt') # Raises Errno::EEXIST.
+  #
+  # When +src+ and +dest+ are both paths to directories,
```

Comment:
This behavior when dest is a directory is the same when src is a file or a directory.  I don't think we need a separate example block for it, maybe just update the above text to mention that it applies both when src is a file and when it is a directory.

Reply:
Cannot add  this to the relevant list item above b/c the covering text says "When +src+ is the path to an existing file:".

Reply:
Then consider reorganizing the sections to avoid the repetition.

Reply:
Done.  One section now, with examples with src as both file and directory.